instrumentation-api-semconv/src/main/java/io/opentelemetry/instrumentation/api/instrumenter/url/UrlAttributesGetter.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.api.instrumenter.url;
+
+import javax.annotation.Nullable;
+
+/**
+ * An interface for getting URL attributes.
+ *
+ * <p>Instrumentation authors will create implementations of this interface for their specific
+ * library/framework. It will be used by the {@link UrlAttributesExtractor} (or other convention
+ * specific extractors) to obtain the various URL attributes in a type-generic way.
+ */
+public interface UrlAttributesGetter<REQUEST> {
+
+  /**
+   * Returns the absolute URL describing a network resource according to <a
+   * href="https://www.rfc-editor.org/rfc/rfc3986">RFC3986</a>.
+   *
+   * <p>Examples: {@code https://www.foo.bar/search?q=OpenTelemetry#SemConv}; {@code //localhost}
+   */
+  @Nullable
+  default String getFullUrl(REQUEST request) {
+    return null;
+  }
+
+  /**
+   * Returns the <a href="https://www.rfc-editor.org/rfc/rfc3986#section-3.1">URI scheme</a>
+   * component identifying the used protocol.
+   *
+   * <p>Examples: {@code https}, {@code ftp}, {@code telnet}
+   */
+  @Nullable
+  default String getUrlScheme(REQUEST request) {
+    return null;
+  }
+
+  /**
+   * Returns the <a href="https://www.rfc-editor.org/rfc/rfc3986#section-3.3">URI path</a>
+   * component.
+   *
+   * <p>Examples: {@code /search}
+   */
+  @Nullable
+  default String getUrlPath(REQUEST request) {
```

Comment:
Should we also have a target? a low cardinality attribute for things like:
`/search/{something}

Reply:
For HTTP semconv, `http.route` contains that kind of information.

In general, these classes are meant to reflect the [semantic conventions](https://github.com/open-telemetry/semantic-conventions/blob/main/semantic_conventions/url.yaml) -- if you want to introduce a new attribute, you might want to start there.